cou/utils/juju_utils.py

```diff
@@ -290,6 +267,20 @@ async def _get_application(self, name: str) -> Application:
 
         return app
 
+    def _get_machine_apps(self, machine_id: int) -> tuple[tuple[str, str], ...]:
+        """Get the applications and charm names deployed in a machine.
+
+        :param machine_id: Machine id.
+        :type machine_id: int
+        :return: Tuple of tuple contains app name and charm name.
+        :rtype: tuple[tuple[str, str], ...]
+        """
+        return tuple(
+            (str(unit.application), str(self._model.applications[unit.application].charm_name))
```

Comment:
Using application name in #247 will help to not making nested for loop, and charm will be potentially used. @gabrielcocenza Can tell you more, since he requested it.

Reply:
After discussing in sync, `charm_name` doesn't seem to be useful at this stage, so I'm voting to remove it from the tuple.

Reply:
done